trin-history/src/storage.rs

```diff
@@ -342,30 +347,38 @@ impl HistoryStorage {
                 HistoryContentKey::try_from(bytes).map_err(ContentStoreError::ContentKey)
             })
             .collect();
-
+        self.metrics.stop_storage_process_timer(timer);
         match result?.first() {
             Some(val) => Ok(Some(val.into())),
             None => Ok(None),
         }
     }
 
-    /// Public method for looking up a content value by its content id
-    pub fn lookup_content_value(&self, id: [u8; 32]) -> anyhow::Result<Option<Vec<u8>>> {
+    /// Internal method for looking up a content value by its content id
+    fn lookup_content_value(&self, id: [u8; 32]) -> anyhow::Result<Option<Vec<u8>>> {
```

Comment:
> If something fails nothing is recorded. We have logs for that.

Can you elaborate on that here? If `lookup_content_value()` fails, then it's not obvious to me where we're emitting these logs. Also, just to clarify, if the process timer is **never** stopped, then nothing is recorded?

Reply:
![image](https://github.com/ethereum/trin/assets/31669092/1b42b77c-4867-474e-825c-5ea37e2bcd6b)
> Also, just to clarify, if the process timer is never stopped, then nothing is recorded?

To my knowledge yes

Reply:
This error then gets forwarded to the overlay which then reports the error

Reply:
> This timer can be stopped and observed at most once, either automatically (when it goes out of scope) or manually. Alternatively, it can be manually stopped and discarded in order to not record its value.

My understanding from the [docs](https://docs.rs/prometheus/latest/prometheus/struct.HistogramTimer.html) is that if it goes out of scope (eg. what happens in this case when `lookup_content_value()?` fail) then it will record the time when it is dropped and count it as an observation. Seems like we probably need to handle these cases and use `stop_and_discard()` on failures? Otherwise we'll end up with false observations skewing the histogram?

Reply:
Good catch I should I should have read the documentation a bit more :laughing:.

I just pushed a fix for this issue by implementing a CustomHistogramTimer which on drop does `stop_and_discard()` instead of `stop_and_record()`.

The reason I choose to fix the problem in this way, is because I did a lot of thinking and came to the conclusion this was the cleanest way to achieve our goal well staying within the structural designs of our project in relation to how we handle metrics